### PR TITLE
Eng 10001 sqlcov rank

### PR DESCRIPTION
**Pull request by johnTcrawford**
Added SqlCoverage tests of the RANK function

### CODE REVIEW

File: tests/scripts/SQLGenerator.py [hunk 1]
```diff
@@ -1033,7 +1033,19 @@ def scan_for_macros(self):
         lines_out = []
         self.__macros = {}
         self.__generators = {}
+        previous_continued_line = ''
         for line in self.__lines:
+            # Allow the use of '\' as a line-continuation character
```
Comment (paulmartel) [inline tests/scripts/SQLGenerator.py]: nice!

**Comment by paulmartel** [comment]
Looks good. Nice incremental framework improvements.

File: tests/scripts/examples/sql_coverage/analytic-template.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+-- Define expressions and clauses used to test RANK, below
+{_columnorexpr |= "_variable"}
+{_columnorexpr |= "@onefun(_variable[@comparabletype])@plus10"}
+{_orderbycolumnorexpr |= "_variable[#ord @rankorderbytype]"}
+{_orderbycolumnorexpr |= "@onefun(_variable[#ord @rankorderbytype])@plus10"}
+{@orderbyclause = "ORDER BY _orderbycolumnorexpr _sortorder"}
+
+-- Test the RANK (Window / SQL Analytic) function
+SELECT               @star,  RANK() OVER (PARTITION BY __[#ord]                                    @orderbyclause) RANK            FROM @fromtables W01
```
Comment (BillWhite) [inline tests/scripts/examples/sql_coverage/analytic-template.sql]: Do you test with no PARTITION BY clause?

**Comment by BillWhite** [comment]
I can't claim to understand all of this.  But it seems plausible.  Except for the one comment about not testing rank() over (...) with no partition by clause it seems complete enough.  There's not much there in rank().

File: tests/scripts/examples/sql_coverage/analytic-template.sql (hunk 2 above)
Comment (johnTcrawford) [inline tests/scripts/examples/sql_coverage/analytic-template.sql]: No, I had no idea that was possible. I'll add such tests, quickly.

**Pull request merged**